studyzen/Set/FlashCardSetController.cs

```diff
@@ -0,0 +1,97 @@
+using Microsoft.AspNetCore.Mvc;
+using StudyZen.Common;
+using StudyZen.FlashCards.Requests;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+
+
+namespace StudyZen.FlashCardSets;
+
+[ApiController]
+[Route("[controller]")]
+public sealed class FlashCardSetController : ControllerBase
+{
+    private readonly IFlashCardSetService _flashCardSetService;
+
+    public FlashCardSetController(IFlashCardSetService flashCardSetService)
+    {
+        _flashCardSetService = flashCardSetService;
+    }
+
+    
+    [HttpPost("add-flashcardset")]
+        public IActionResult AddFlashCardSet([FromBody] CreateFlashCardSetRequest? request)
+        {
+            request = request.ThrowIfRequestArgumentNull(nameof(request));
+
+            var flashCardSetId = _flashCardSetService.AddFlashCardSet(request);
+
+            var response = new
```

Comment:
as previously mentioned with flashCard, `AddFlashCardSet` could return not the id, but the whole object to avoid duplicating the same code and creating object twice

Reply:
Fixed